provides.py

```diff
@@ -65,6 +65,15 @@ def set_role_id(self, unit, role_id, token):
         # for cmr we will need to the other end to provide their unit name
         # expicitly.
         unit_name = self.get_remote_unit_name(unit)
+
+        # each unit in this relation should have one role_id, token pair associated
+        # if the unit_name shared over the relation is updated, we should clear
+        # previously associated role_id/token pairs from the relation
+        unit_id = unit_name.split("/")[1]  # the unit's id number
+        removable = f"{unit_id}_role_id", f"{unit_id}_token"
+        for key in unit.relation.to_publish.keys():
+            if any(key.endswith(r) for r in removable):
```

Comment:
clever support for both old short and new long reqs coming from remote units :heart:

Reply:
this will only occur when the vault charm pulls in this updated interface.  Until then there will be some old token/role_id pairs in the relations shared back with the requires side.